app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/agent/BusinessAccountingMethodController.scala

```diff
@@ -77,14 +77,16 @@ class BusinessAccountingMethodController @Inject()(businessAccountingMethod: Bus
           formWithErrors =>
             Future.successful(BadRequest(view(formWithErrors, id, isEditMode))),
           businessAccountingMethod =>
-            incomeTaxSubscriptionConnector.saveSubscriptionDetails(reference, businessAccountingMethodKey, businessAccountingMethod) map { _ =>
+            incomeTaxSubscriptionConnector.saveSubscriptionDetails(reference, businessAccountingMethodKey, businessAccountingMethod).map(_ =>
               (id, isEditMode, isEnabled(SaveAndRetrieve)) match {
-                case (Some(id), _, true) =>
-                  Redirect(routes.SelfEmployedCYAController.show(id, isEditMode = isEditMode))
+                case (Some(id), _, _) =>
+                  Redirect(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.agent.routes.SelfEmployedCYAController.show(id, isEditMode = isEditMode))
+                case (_, true, false) =>
+                  Redirect(appConfig.subscriptionFrontendClientFinalCYAController)
                 case _ =>
-                  Redirect(s"${appConfig.subscriptionFrontendClientRoutingController}?editMode=$isEditMode")
+                  Redirect(appConfig.subscriptionFrontendClientRoutingController)
```

Comment:
It looks like you can also leave this part:
```
case (_, true, false) =>
  Redirect(appConfig.subscriptionFrontendClientFinalCYAController)
case _ =>
  Redirect(appConfig.subscriptionFrontendClientRoutingController)
```
Like this:
```
case _ =>
  Redirect(s"${appConfig.subscriptionFrontendClientRoutingController}?editMode=$isEditMode")
```
Because the agent RoutingController handles the editMode logic; unlike the individual RoutingController.